cuda_core/cuda/core/experimental/_device.py

```diff
@@ -14,6 +14,1021 @@
+        handle_type_win32_kmt_handle_supported (bool): True if device supports exporting memory to a Win32 KMT handle with cuMemExportToShareableHandle, False if not.
+        max_blocks_per_multiprocessor (int): Maximum number of thread blocks that can reside on a multiprocessor.
+        generic_compression_supported (bool): True if device supports compressible memory allocation via cuMemCreate, False if not.
+        max_persisting_l2_cache_size (int): Maximum L2 persisting lines capacity setting in bytes.
+        max_access_policy_window_size (int): Maximum value of CUaccessPolicyWindow::num_bytes.
+        gpu_direct_rdma_with_cuda_vmm_supported (bool): True if device supports specifying the GPUDirect RDMA flag with cuMemCreate, False if not.
+        reserved_shared_memory_per_block (int): Amount of shared memory per block reserved by CUDA driver in bytes.
+        sparse_cuda_array_supported (bool): True if device supports sparse CUDA arrays and sparse CUDA mipmapped arrays, False if not.
+        read_only_host_register_supported (bool): True if device supports using the cuMemHostRegister flag CU_MEMHOSTERGISTER_READ_ONLY to register memory that must be mapped as read-only to the GPU, False if not.
+        memory_pools_supported (bool): True if device supports using the cuMemAllocAsync and cuMemPool family of APIs, False if not.
+        gpu_direct_rdma_supported (bool): True if device supports GPUDirect RDMA APIs, False if not.
+        gpu_direct_rdma_flush_writes_options (int): The returned attribute shall be interpreted as a bitmask, where the individual bits are described by the CUflushGPUDirectRDMAWritesOptions enum.
+        gpu_direct_rdma_writes_ordering (int): GPUDirect RDMA writes to the device do not need to be flushed for consumers within the scope indicated by the returned attribute.
+        mempool_supported_handle_types (int): Bitmask of handle types supported with mempool based IPC.
+        deferred_mapping_cuda_array_supported (bool): True if device supports deferred mapping CUDA arrays and CUDA mipmapped arrays, False if not.
+        numa_config (int): NUMA configuration of a device.
+        numa_id (int): NUMA node ID of the GPU memory.
+        multicast_supported (bool): True if device supports switch multicast and reduction operations, False if not.
+    """
+
+    def __init__(self):
+        raise RuntimeError("DeviceProperties should not be instantiated directly")
+
+    slots = "_handle"
+
+    def _init(handle):
+        self = DeviceProperties.__new__(DeviceProperties)
+        self._handle = handle
+        return self
+
+    @property
+    def max_threads_per_block(self):
+        return handle_return(
+            driver.cuDeviceGetAttribute(
+                driver.CUdevice_attribute.CU_DEVICE_ATTRIBUTE_MAX_THREADS_PER_BLOCK, self._handle
+            )
+        )
+
+    @property
+    def max_block_dim_x(self):
```

Comment:
Since the docstring for this class to so long, you won't be able to see the description of the property you are interested in if your IDE shows docstrings on mouse hover? Why not include the docstrings for each of the properties here?